packages/mui-material/src/useAutocomplete/useAutocomplete.js

```diff
@@ -247,21 +257,22 @@ function useAutocomplete(props) {
 
   const listboxAvailable = open && filteredOptions.length > 0 && !readOnly;
 
-  const focusTag = useEventCallback((tagToFocus) => {
-    if (tagToFocus === -1) {
+  const focusItem = useEventCallback((itemToFocus) => {
+    if (itemToFocus === -1) {
       inputRef.current.focus();
     } else {
-      anchorEl.querySelector(`[data-tag-index="${tagToFocus}"]`).focus();
+      const indexType = renderValue ? 'data-item-index' : 'data-tag-index';
```

Comment:
Please add a comment explaining that `data-tag-index` exists for backward compatibility. It's not immediately clear what's going on here without knowing the context of this PR.

Reply:
Added.